lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -142,7 +142,12 @@ def rebuild_inventory(input)
         action_inputs = input.values.map { |hash| hash[:input][:action_input] }
         hostnames = action_inputs.map { |hash| hash[:name] }
         inventories = action_inputs.map { |hash| hash[:ansible_inventory] }
-        host_vars = inventories.map { |i| i['_meta']['hostvars'] }.reduce(&:merge)
+        host_vars = inventories.map { |i| i['_meta']['hostvars'] }.reduce({}) do |acc, hosts|
+          hosts.reduce(acc) do |inner_acc, (hostname, vars)|
+            vars[:ansible_ssh_private_key_file] ||= ForemanRemoteExecutionCore.settings[:ssh_identity_key_file]
+            inner_acc.merge(hostname => vars)
+          end
+        end
```

Comment:
Do we have to fill the default here for every host?
Wouldn't it be better to do this globally for the whole inventory and leave the per host one blank?

Reply:
iirc currently the key is defined for each host, but the value is optionally nil. We'd have to stop sending it over if the value is nil. Not saying that's bad, but it would probably result in a bigger diff

Reply:
Agreed https://projects.theforeman.org/issues/32595.